### DISCUSSION TITLE
Update points with button click

### DISCUSSION THREAD

**Discussion by MuslemRahimi** [pocketbase/pocketbase#2235] [Q&A]
Hi,

Setup: 

I have a collection called "posts" where user can post content. Each post has the key value "vote". In my sveltekit app in the file +page.svelte i have two buttons where user can click to upvote (+1) or downvote (-1) the post. 

How can i send the data to +page.server.ts to update the database.
As an argument i can sent the postId = 'pv8t9om6nqb4mgo'
`await locals.pb.collection("posts").update(postId, {
            "vote+": 1,
})`

Constrain:
I want that each user can only upvote or dowvote each post once. How can I achieve this?

Any help is much much appreciated

**Comment by ganigeorgiev** [comment]
You'll have to store somewhere each individual user vote, so a simple "vote" number field would not suffice in this case.

Instead you can create a new collection `postVotes` with at least 3 fields - "post" (single `relation` to the `posts` collection), "user" (single `relation` to the `users` collection) and "action" as optional `bool` field to indicate whether it was an upvote or downvote.
Then you can define a "Unique index constraint" with the 2 relation fields to ensure that all user-post pairs are unique (_this is available with v0.14.x in the Collection settings > Fields tab_).

To fetch all posts votes you can then send:

```js
const votes = await pb.collection("postVotes").getFullList({ filter: `post="${postId}"` })
```

OR alternatively, you can also create a separate "View collection" with a SQL query like:

```sql
-- assuming that "action" is a bool field in the postVotes collection
SELECT
    postVotes.post as id,
    SUM(CASE action WHEN 1 THEN 1 ELSE 0 END) totalUpvotes
    SUM(CASE action WHEN 0 THEN 1 ELSE 0 END) totalDownvotes
FROM postVotes
GROUP BY postVotes.post
```
_I haven't tested the above query so you may have to adjust it. It should create a read-only collection with 3 fields - "id" (the post id), "totalUpvotes" and "totalDownvotes"._

**Reply by MuslemRahimi** [reply]
Is it possible to do something like this here to find the specifc postId with the userId:


let currentVote = await locals.pb.collection("alreadyVoted").getList(1, 50, {
            filter: `user="${user}"`,
            filter:`post="${postId}"`,
        })`
`

I want to extract that specific element of the collection for the user

**Reply by ganigeorgiev** [reply]
You can try with:

```js
await locals.pb.collection("alreadyVoted").getList(1, 50, {
    filter: `user="${user}" && post="${postId}"`,
}) 
```

**Reply by MuslemRahimi** [reply]
thanks. That does not give me an error but it does not solve the issue unfortunately.

When i type the above code i don't get the specific user and postId in the collection 'alreadyVoted'. Instead i get the postId and another user that already exist. 

For example 
`{
  page: 1,
  perPage: 50,
  totalItems: 1,
  totalPages: 1,
  items: [
    n {
      collectionId: 'ofr9fkr5a3zpz91',
      collectionName: 'alreadyVoted',
      created: '2023-04-07 10:26:03.588Z',
      id: 'zpwjw6691ukjd8t',
      post: '4q9kh7gpapolv1l',
      type: 'upvote',
      updated: '2023-04-07 10:30:18.464Z',
      user: 'keaj642y5u629u6',
      expand: {}
    }
  ]
}
`

But i have 
`user = 'db5s41oatgoeh0q' and postId = '4q9kh7gpapolv1l'`

`await locals.pb.collection("alreadyVoted").getList(1, 50, {
    filter: `user="${user}" && post="${postId}"`,
}) `

In my case the user ''db5s41oatgoeh0q' does not exist yet and i want to return a empty list of items

**Reply by MuslemRahimi** [reply]
I hope this helps too

![Screenshot from 2023-04-07 12-46-56](https://user-images.githubusercontent.com/22890183/230596159-f904e13a-8163-4129-b485-c49907c966be.png)
![Screenshot from 2023-04-07 12-47-01](https://user-images.githubusercontent.com/22890183/230596166-bb85fdbd-9c59-4ef3-b588-86801767915a.png)
